L1Trigger/Phase2L1ParticleFlow/src/corrector.cc

```diff
@@ -0,0 +1,193 @@
+#include "L1Trigger/Phase2L1ParticleFlow/src/corrector.h"
+
+#include <iostream>
+#include <sstream>
+#include <cstdio>
+#include <cstdlib>
+#include <cassert>
+#include <unordered_map>
+#include <TFile.h>
+#include <TKey.h>
+#include <TH1.h>
+#include <TH2.h>
+#include <TAxis.h>
+#include "FWCore/Utilities/interface/CPUTimer.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+
+#include "DataFormats/L1TParticleFlow/interface/PFCluster.h"
+
+l1tpf::corrector::corrector(const std::string &filename, float emfMax, bool debug) : emfMax_(emfMax) {
+  if (!filename.empty())
+    init_(filename, "", debug);
+}
+l1tpf::corrector::corrector(const std::string &filename, const std::string &directory, float emfMax, bool debug)
+    : emfMax_(emfMax) {
+  if (!filename.empty())
+    init_(filename, directory, debug);
+}
+
+l1tpf::corrector::corrector(TDirectory *src, float emfMax, bool debug) : emfMax_(emfMax) { init_(src, debug); }
+
+void l1tpf::corrector::init_(const std::string &filename, const std::string &directory, bool debug) {
```

Comment:
methods should not end in underscore

Reply:
says who ?

Reply:
leads to confusion between member variables (which should end in underscore)

Reply:
I think you are mistaken: member variables end with an underscore to make it easy to distinguish them from global and local variables, not to distinguish them from methods (or vice versa).

There rarely is confusion between a data member (a variable) and a method (a function).
In fact, names ending with an underscore are sometimes (often? did not do a full `git grep`) used for private methods that implement a public method with a similar name.

For some example, see e.g. the files under `FWCore/Framework/`.

Reply:
it also helps to distinguish member variables from accessor methods.

but you're right that in this case, as a private method, it should be fine.